src/home/user-record-util.c

```diff
@@ -1404,6 +1404,208 @@ int user_record_is_supported(UserRecord *hr, sd_bus_error *error) {
         return 0;
 }
 
+static int remove_safe_json_fields_regular(JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *blobs = NULL;
+        int r;
+
+        assert(v);
+
+        char **safe_fields = STRV_MAKE(
+                /* For display purposes */
+                "realName",
+                "emailAddress", /* Just the $EMAIL env var */
+                "iconName",
+                "location",
+
+                /* Basic account settings */
+                "shell",
+                "umask",
+                "environment",
+                "timeZone",
+                "preferredLanguage",
+                "additionalLanguages",
+
+                /* Authentication methods */
+                "pkcs11TokenUri",
+                "fido2HmacCredential",
+                "recoveryKeyType",
+
+                "lastChangeUSec", /* Necessary to be able to change record at all */
+                "lastPasswordChangeUSec" /* Ditto, but for authentication methods */
+        );
```

Comment:
this makes a ton of sense, but I'd prefer if this was configurable in the user records themselves. i.e. add a new field with a string array: `ownerModifiableFields` or so.

And if the field is not set, define a good default allowlist like this one, but hide that in a user_record_owner_modifiable_fields() helper

Reply:
:+1: 

That idea covers the regular fields and perMachine fields. What about privileged? Should we have an `ownerModifiableFields` sub-field in the `privileged` section that does the same as the `regular`/`perMachine` one? Should we have just that one that handles field names in the `privileged` section too?

Reply:
please suffix list with a comma